js/dragonex.js

```diff
@@ -0,0 +1,912 @@
+            const sells = [];
+            sells.push (this.safeInteger (this.safeValue (response, 'data', {})['sells'][i], 'price', 0));
+            sells.push (this.safeInteger (this.safeValue (response, 'data', {})['sells'][i], 'volume', 0));
+            sellsList.push (sells);
+        }
+        orderBook['bids'] = buysList;
+        orderBook['asks'] = sellsList;
+        return orderBook;
+    }
+
+    parseTicker (ticker, market = undefined) {
+        ticker = this.safeValue (this.safeValue (ticker, 'data', {}), 'list', [])[0];
+        const lastNum = ticker.length - 1;
+        const lastFourNum = ticker.length - 4;
+        const timestamp = ticker[lastNum];
+        const last = parseFloat (ticker[3]);
+        return {
+            'symbol': market['symbol'],
+            'timestamp': timestamp,
+            'datetime': this.iso8601 (timestamp),
+            'high': ticker[4],
+            'low': ticker[5],
+            'bid': ticker[6],
+            'bidVolume': undefined,
+            'ask': ticker[7],
+            'askVolume': undefined,
+            'vwap': undefined,
+            'open': ticker[2],
+            'close': last,
+            'last': last,
+            'previousClose': undefined,
+            'change': ticker[8],
+            'percentage': ticker[9],
+            'average': (last + parseFloat (ticker[2])) / 2,
+            'baseVolume': ticker[lastFourNum],
+            'quoteVolume': undefined,
+            'info': ticker,
+        };
+    }
+
+    async fetchTicker (symbol, params = {}) {
+        await this.loadMarkets ();
+        const market = this.market (symbol);
+        const request = {
+            'symbol_ids': market['symbol_id'],
+        };
+        const response = await this.aicoinGetMarketReal (this.extend (request, params));
+        return this.parseTicker (response, market);
+    }
+
+    parseOHLCV (ohlcv, market = undefined, timeframe = '1m', since = undefined, limit = undefined) {
+        return [
+            ohlcv[0],
+            parseFloat (ohlcv[1]),
+            parseFloat (ohlcv[2]),
+            parseFloat (ohlcv[3]),
+            parseFloat (ohlcv[4]),
+            parseFloat (ohlcv[5]),
+        ];
+    }
+
+    async fetchOHLCV (symbol, timeframe = '1m', since = undefined, limit = undefined, params = {}) {
+        await this.loadMarkets ();
+        const market = this.market (symbol);
+        const request = {
+            'symbol_id': market['symbol_id'],
+            'kline_type': this.timeframes[timeframe],
+        };
+        if (since !== undefined) {
+            since = 0;
+            request['st'] = since;
+        }
+        if (limit !== undefined) {
+            request['count'] = limit;
+        }
+        const response = await this.v1GetMarketKline (this.extend (request, params));
+        const lists = this.safeValue (this.safeValue (response, 'data'), 'lists', []);
+        const ohlcvs = [];
+        for (let i = 0; i < lists.length; i++) {
```

Comment:
We don't need this loop, because the parsing is done in `parseOHLCVs` and `parseOHLCV`, see how this is done with other exchanges.